src/rev_ai/custom_vocabularies_client.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+"""Speech recognition tools for using Rev.ai"""
+
+from .baseclient import BaseClient
+
+try:
+    from urllib.parse import urljoin
+except ImportError:
+    from urlparse import urljoin
+
+
+class RevAiCustomVocabulariesClient(BaseClient):
+    """Client which implements Rev.ai CustomVocabulary API"""
+
+    def __init__(self, access_token):
+        """Constructor
+
+        :param access_token: access token which authorizes all requests and
+                             links them to your account. Generated on the
+                             settings page of your account dashboard
+                             on Rev.ai
+        """
+        BaseClient.__init__(self, access_token)
+
+        self.base_url = urljoin(self.base_url, 'vocabularies/')
+
+    def submit_custom_vocabularies(
+            self,
+            custom_vocabularies,
+            callback_url=None,
+            metadata=None):
+        """Submit custom vocabularies.
+
+        :param custom_vocabularies: List of CustomVocabulary objects
+        :param callback_url: callback url to invoke on job completion as a
+                             webhook
+        :param metadata: info to associate with the transcription job
+        """
+
+        if not custom_vocabularies:
+            raise ValueError('custom_vocabularies must be provided')
+
+        payload = self._create_custom_vocabularies_options_payload(
+            custom_vocabularies,
+            callback_url,
+            metadata
+        )
+
+        response = self._make_http_request(
+            "POST",
+            self.base_url,
+            json=payload
+        )
+
+        return response.json()
+
+    def get_custom_vocabularies(
+            self,
+            id):
+        """ Get the custom vocabulary status
+        :param id: string id of custom vocabulary submission
+        """
+
+        response = self._make_http_request(
+            "GET",
+            urljoin(self.base_url, "{}".format(id))
+        )
+
+        return response.json()
+
+    def _create_custom_vocabularies_options_payload(
```

Comment:
I think it's somewhat unnecessary to separate this into another method. This method isn't going to be used anywhere else and it doesn't make the caller much less complex

It's somewhat preference though. How it is isn't bad

Reply:
It makes it better for understanding of the http request structure as well as being more modular which makes the code easier to understand in my opinion. It also parallels the pattern in `apiclient` which seems nice, but is no solid reason on its own.